core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -1516,6 +1516,13 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.ENFORCE)
           .setScope(Scope.MASTER)
           .build();
+  public static final PropertyKey MASTER_METASTORE_BLOCK_ITERATOR_READAHEAD_SIZE =
+      new Builder(Name.MASTER_METASTORE_BLOCK_ITERATOR_READAHEAD_SIZE)
+          .setDefaultValue("10000")
+          .setDescription("The read-ahead size for block iterator.")
```

Comment:
Can you describe this more, and the implications of a small one and a large one?

Also, what is "size"? rows? bytes?

Reply:
bytes

Reply:
I see. Could you update the description to mention bytes, and maybe some of the implications of the size? For example, if we have a billion blocks, is reading 4mb at a time efficient enough? Should this be a larger size?

Reply:
I guess 4MB suffices, as it'd prefetch about 10K inodes and 100K block-Ids. This should be enough as iterator content are ultimately written back to disk.